toolchain/ltrace.py

```diff
@@ -69,15 +69,22 @@ def setUp(self):
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel("Fail to install %s required for this test." %
                             package)
-
-        # Source: git clone git://git.debian.org/git/collab-maint/ltrace.git
-        git.get_repo('git://git.debian.org/git/collab-maint/ltrace.git',
+        run_type = self.params.get("type", default="upstream")
+        if run_type == "upstream":
+            # Source: git clone git://git.debian.org/git/collab-maint/ltrace.git
+            git.get_repo('git://git.debian.org/git/collab-maint/ltrace.git',
                      destination_dir=os.path.join(self.srcdir, 'ltrace'))
 
-        self.src_lt = os.path.join(self.srcdir, "ltrace")
-        os.chdir(self.src_lt)
-        process.run('patch -p1 < %s' %
-                    os.path.join(self.datadir, 'ltrace.patch'), shell=True)
+            self.src_lt = os.path.join(self.srcdir, "ltrace")
+            os.chdir(self.src_lt)
+            process.run('patch -p1 < %s' %
+                        os.path.join(self.datadir, 'ltrace.patch'), shell=True)
```

Comment:
@harish-24 Does it always true, having ltrace.patch?,what if not it would be better to handle with a flag so that direct git HEAD can be tested. and how about moving the git link to config(yaml) so that custom gits can be tested, applicable for all such occurrences in other tests too..

Reply:
@sathnaga ltrace.patch is temp fix for compilation issue of test suite , i already submitted patch against ltrace  source , when it got accepted we can remove from test code

Reply:
@sathnaga Sure, I will move the url to config.